content/automate/ManageIQ/Service/Generic/StateMachines/GenericLifecycle.class/__methods__/update_status.rb

```diff
@@ -46,24 +46,34 @@ def service_action
                 end
               end
 
-              def update_status(service)
-                # Get status from input field status
-                status = @handle.inputs['status']
+              def error
+                @handle.root['ae_result'] == "error"
+              end
 
+              def updated_message(service, status)
                 updated_message = "Server [#{@handle.root['miq_server'].name}] "
                 updated_message += "Service [#{service.name}] #{service_action} "
                 updated_message += "Step [#{@handle.root['ae_state']}] "
                 updated_message += "Status [#{status}] "
                 updated_message += "Current Retry Number [#{@handle.root['ae_state_retries']}]"\
                                     if @handle.root['ae_result'] == 'retry'
                 @handle.log(:info, "Status message: #{updated_message} ")
-                update_task(updated_message, status)
+                updated_message
+              end
 
-                if @handle.root['ae_result'] == "error"
-                  @handle.create_notification(:level   => "error",
-                                              :subject => service,
-                                              :message => "Generic Service Error: #{updated_message}")
-                end
+              def error_handling(message)
+                @handle.create_notification(:level   => "error",
+                                            :subject => service,
+                                            :message => "Generic Service Error: #{message}")
+                service.on_error(service_action)
+              end
+
+              def update_status
+                # Get status from input field status
+                status = @handle.inputs['status']
+
+                update_task(updated_message(service, status), status)
```

Comment:
@tinaafitz 
We would be calling the service function multiple times here, in case of error there are 3 method calls.
Maybe we can do this in a separate PR. We can either use a @service attribute in the method and just use it everywhere, so we dont have to pass it around or call the function multiple times